### PR TITLE
Simplify merge with master

### PR DESCRIPTION
**Pull request by 0xPoly**
Working so far: Installing and Listing plugins, as well as executing binaries.

### CODE REVIEW

**Comment by bassosimone** [comment]
Hi @0xPoly,

I'd be happy to test this and merge. To facilitate the merge, could you please rebase to master?

Regarding tests, any suggestion on how I could test this?

Thanks!

**Comment by 0xPoly** [comment]
Hey.

Are you asking about testing the features implemented here or how we should implement tests in general?

**Comment by 0xPoly** [comment]
That was a bit tricky, but rebase complete :)

File: plugins/ooni/main.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+    "name": "OONI",
+    "short": "Open Observatory of Network Interference",
+    "description": "A free software, global observation network for detecting censorship, surveillance and traffic manipulation on the internet",
+    "version": "1.0.1",
+    "exec": [
+    {
+        "full-path": "/usr/bin/traceroute"
```
Comment (hellais) [inline plugins/ooni/main.json:8]: I guess these are just examples at the moment, but the binary could at least be the correct one

File: docs/design.md [hunk 2]
```diff
@@ -1,98 +0,0 @@
-Network Meter Design Document
```
Comment (hellais) [inline docs/design.md:1]: Why is the design document removed?

I think it's still relevant to keep.

File: plugins/ooni/main.json (hunk 1 above)
Comment (0xPoly) [inline plugins/ooni/main.json:8]: The program doesn't currently make any use of the binary's paths, so I haven't changed them yet. Just wanted to tested if multiple installed plugins were displayed properly.

File: docs/design.md (hunk 2 above)
Comment (0xPoly) [inline docs/design.md:1]: Fixed.

**Comment by bassosimone** [comment]
@0xPoly, I meant to ask how I could run network-meter so to better understand the changes made in this pull request by running the code, apologies if the question caused confusion. (On a related note, we will certainly need tests in future, but I'd say now it's early to start working on tests.)

**Comment by 0xPoly** [comment]
Checkout the branch and run:

npm install -d
npm install
npm start

**Comment by bassosimone** [comment]
Thanks!

File: package.json [hunk 3]
```diff
@@ -1,13 +1,7 @@
 {
   "name": "network-meter",
-  "productName": "Network Meter",
   "version": "0.1.0",
-  "description": "Graphical frontend for various network tests",
   "main": "index.js",
-  "repository": {
-      "type": "git",
-      "url" : "https://github.com/hellais/network-meter"
-  },
```
Comment (bassosimone) [inline package.json:10]: I think we would like to keep "productName", "description" and "repository" above, no?

File: plugin-parser.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    
+    fs.readdir(pluginFolder, function(err, files) {
+        if (err) {
+            console.log(err);
+            return;
+        }
+
+        // consider revamp to async forEach.
+        // this method causes considerable startup delay
+        var callbackCounter = files.length;
```
Comment (bassosimone) [inline plugin-parser.js:16]: I agree that we can improve this, but I think it can be done after the code here is in tree.

**Comment by bassosimone** [comment]
Having read the diff and tested the program, I'd say let's merge this and make further changes we need in tree. Just after merging, I'll add all the things that were notes as comments of this pull request, plus a couple more ones I noticed while testing as issues, so we do not forget about them :-).

Let's move forward! :+1:

**Comment by bassosimone** [comment]
Done!

**Pull request merged**